src/scenekit.cs

```diff
@@ -1340,6 +1378,83 @@ interface SCNLight : SCNAnimatable, SCNTechniqueSupport, NSCopying, NSSecureCodi
 		[Export ("lightWithMDLLight:")]
 		SCNLight FromModelLight (MDLLight mdllight);
 #endif
+
+		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+		[Export ("probeType", ArgumentSemantic.Assign)]
+		SCNLightProbeType ProbeType { get; set; }
+
+		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+		[Export ("probeUpdateType", ArgumentSemantic.Assign)]
+		SCNLightProbeUpdateType ProbeUpdateType { get; set; }
+
+		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+		[Export ("probeExtents", ArgumentSemantic.Assign)]
+		Vector3 ProbeExtents {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			set;
+		}
+
```

Comment:
Noob question - What does the `MarshalDirective` do here? Does `NativePrefix = "xamarin_simd__"` mean that this binding has something to do AOT compilation?

Reply:
`MarshalDirective` is a workaround we have to work around a calling convention limitation with P/Invokes.

The problem is that certain native types (such as `Vector3`) are not put in the normal registers when used as parameters to methods, instead they're passed in special SIMD registers. Unfortunately managed code (i.e. P/Invokes) doesn't know about this, so we can't call these native methods using P/Invokes because the arguments would be in the wrong registers.

Our workaround is a bit involved: we generate a native wrapper method where the parameters are passed in the expected registers, and that calls the actual native function, so that the native compiler puts the arguments in the right registers. The `MarshalDirective` attribute tells the generator that it should generate a call to the wrapper method instead of the normal P/Invoke (and that the name of the wrapper method is the name of the native function prefixed by `xamarin_simd__`, and the library where the wrapper method is located - `__Internal` means the executable itself).

Reply:
🤯 (_takes it in slowly_)
Thanks! 😃